### DISCUSSION TITLE
custom link extension

### DISCUSSION THREAD

**Discussion by pimousse1099** [ueberdosis/tiptap#259] [General Questions]
Hi,
I'm trying to create a customLink Mark in order to get the following code when I add a link:

```
<ressource
  data-type="external-link"
  data-href="https://www.goodplanet.org/fr"
  data-text="https://www.goodplanet.org/fr"
  data-_target=""/>
    <a href="https://www.goodplanet.org/fr">https://www.goodplanet.org/fr</a>
</ressource>
```

I'm trying to start from https://github.com/scrumpy/tiptap/blob/master/packages/tiptap-extensions/src/marks/Link.js but can't figure out what to change in the `parseDOM` & `toDOM`

I've also seen the `get view` function but the problem remains the same ...

**Comment by starostin** [comment]
Hi, do you still need this example? Because I have already created custom node for link preview. It is a bit different from what you need, but the idea is the same.

**Comment by ashleymoogle** [comment]
Up! I'm struggling to do the same thing. The doc is very very VERY bare (inexistant even)

**Comment by mike-seekwell** [comment]
@starostin Do you still have this? Mind sharing it?

**Comment by hanspagel** [comment]
@starostin would you mind to share your custom node?

**Reply by justinmoon** [reply]
@starostin I'd also love to see this custom node!

**Comment by ikasianiuk** [comment]
@starostin could you please share with your solution of custom node for link preview?
